sdk/samples/iot/paho_iot_hub_pnp_component_sample.c

```diff
@@ -310,23 +306,17 @@ static void create_and_configure_mqtt_client(void)
   az_iot_pnp_client_options options = az_iot_pnp_client_options_default();
   options.component_names = pnp_device_components;
   options.component_names_length = pnp_components_length;
-  rc = az_iot_pnp_client_init(
-      &pnp_client, env_vars.hub_hostname, env_vars.hub_device_id, model_id, &options);
-  if (az_result_failed(rc))
-  {
-    IOT_SAMPLE_LOG_ERROR("Failed to initialize pnp client: az_result return code 0x%08x.", rc);
-    exit(rc);
-  }
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(
```

Comment:
The only reason I would leave some of these as is, is because they match what is in the other samples.  But I can go either way.  Since the macro has been introduced for the pnp component sample, it also makes sense to update all the places that log and exit on an az_result fail.